streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -529,11 +531,13 @@ public void run() {
             cleanRun = true;
         } catch (final KafkaException e) {
             // just re-throw the exception as it should be logged already
+            // TODO: this is not true since task.update() call may throw ProcessorStateException
+            //       which is a KafkaException but it never get logged.
             throw e;
         } catch (final Exception e) {
             // we have caught all Kafka related exceptions, and other runtime exceptions
             // should be due to user application errors
-            log.error("{} Streams application error during processing: ", logPrefix, e);
+            log.error("{} Streams application error during processing", logPrefix, e);
```

Comment:
The reason is that the error and exception trace would actually not be printed in the same line. To just show you an example:

```
[2017-07-25 17:22:10,359] WARN stream-thread [clientId-StreamThread-10] Failed to commit task 0_0 state when suspending:(org.apache.kafka.streams.processor.internals.StreamThread:1180)
Exception
at ...
at ...
```

So it is empty after the colon actually.